### PR TITLE
feat: add pagination example (without prefetching)

### PR DESCRIPTION
**Pull request by trungk18**
## Pagination example

This is to bring the [pagination example](https://tanstack.com/query/v4/docs/examples/react/pagination) from query-v4 docs into #4. 

However, it hasn't yet had the prefetching functionality.

https://user-images.githubusercontent.com/6767322/195976899-572911f9-ba77-4e7d-86ca-c888659236a6.mp4

## Preloading next page

What I have in mind for the prefetching is to bake this code into the ProjectsService and trigger them manually on component init or on the next page

```ts
export class ProjectsService {
  private queryClient = inject(QueryClient);

  preFetchNextPage(page: number, data: PaginatedProject | undefined) {
    if (data?.hasMore) {
      this.queryClient.prefetchQuery(['projects', page + 1], () =>
        getProjects(page + 1)
      );
    }
  }
}
```

```diff
export class PaginationPageComponent {
  triggerFetch(data?: PaginatedProject) {
    this.projects$ = this.projectsService.getProjects(this.page);
+    this.projectsService.preFetchNextPage(this.page, data);
  }

  nextPage(data: PaginatedProject | undefined) {
    this.page = data?.hasMore ? this.page + 1 : this.page;
+    this.triggerFetch(data);
  }
}
```

### CODE REVIEW

File: packages/playground/src/app/pagination-page/pagination-page.component.ts [hunk 1]
```diff
@@ -1,16 +1,69 @@
...
+    </div>
+  `,
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class PaginationPageComponent implements OnInit {
-  constructor() {}
+export class PaginationPageComponent {
+  page = 0;
+  projectsService = inject(ProjectsService);
+  projects$ = this.projectsService.getProjects(this.page);
```
Comment (trungk18) [inline packages/playground/src/app/pagination-page/pagination-page.component.ts:54]: I actually want to the below instead, however I am having a hard time defining type for `projects$` 😂

```
projects$: any // <-- how to extract the correct type here as NgQueryObserverReturnType is not exported
  constructor() {
    this.triggerFetch();
  }

  triggerFetch() {
    this.projects$ = this.projectsService.getProjects(this.page);
  }
```

**Comment by zarghamkhandev** [comment]
oops, seems like we are in conflict. I just created PR for pagination regarding #11

**Comment by NetanelBasal** [comment]
@trungk18 @zarghamkhandev it doesn't work like the original example. In the original one, the data is kept when you move between pages. I'm working on it. It was tricky and required some changes.

**Comment by NetanelBasal** [comment]
@trungk18 @zarghamkhandev checkout https://github.com/ngneat/query/commit/66a27aeef3c3ec6adbc6f87d01d66541cad706af.

It was very tricky. We need to think about a way to make it simpler.

**Comment by trungk18** [comment]
@NetanelBasal Noted, I am reading the code.

@zarghamkhandev Thanks bro, I didn't notice that you work on the same example hehe.
Maybe in the future, should we create a new issue and assign it to ourselves? So that it is visible who is working on which tasks.

**Comment by NetanelBasal** [comment]
@trungk18 @zarghamkhandev do you want to take one of the final issues before going to beta?

https://github.com/ngneat/query/issues

**Comment by trungk18** [comment]
@NetanelBasal I can take #28 to setup the playground.

Regarding #22, could you please give some outline and then we can take it from there? Docs are quite important and we can split the work.

**Comment by NetanelBasal** [comment]
Please comment on the issue so that I can assign it to you.

Regarding the docs, I'll update the issue.

**Pull request closed without merging**